packages/react-charting/src/components/PieChart/PieChart.base.tsx

```diff
@@ -38,15 +39,28 @@ export class PieChartBase extends React.Component<IPieChartProps, IPieChartState
   public render(): JSX.Element {
     const { data, width, height, colors, chartTitle } = this.props;
 
+    const finalColors = [];
     const { theme, className, styles, culture } = this.props;
+
+    if (data) {
+      for (let i = 0, nextColor = 0; i < data.length; i++) {
+        if (colors && i < colors?.length && typeof colors[i] !== 'undefined') {
+          finalColors.push(getColorFromToken(colors[i]));
+        } else {
+          finalColors.push(getNextColor(nextColor, 0, theme?.isInverted));
+          nextColor++;
+        }
+      }
+    }
+
     this._classNames = getClassNames(styles!, {
       theme: theme!,
       width: width!,
       height: height!,
       className,
     });
-    const radius = Math.min(width!, height!) / 2;
-    const outerRadius = radius - 10;
+    const radius = Math.min(width! - 2 * 40, height! - 2 * 16) / 2;
```

Comment:
What does this value represent

Reply:
So the outerRadius for the Pie chart is being computed here, such that the text doesn't end up getting cropped in any case.

The text can have a maxWidth of 40, so we're subtracting it from width (2* for both sides). Then the line height of the text is 16, so we're subtracting it from the height.

This would ensure that the radius is such that even in the most extreme cases.

I'll put some comments here to clarify, and maybe use some appropriately named variables to define the values.